appGeo/src/app/game/game.component.ts

```diff
@@ -89,6 +102,31 @@ export class Game implements OnInit {
 
   }
 
+  updateGameRulesDatabase(data: object) {
+    this.gameRules = data["value"]
+    console.log("updated game rules: " + JSON.stringify(this.gameRules));
+  }
+
+  updateMapDatabase(data: object) {
+    this.map = data["value"]
+    console.log("updated map: " + JSON.stringify(this.map));
+  }
+
+  updateStartTime(data: object) {
+    this.startTime = data["value"]
+    console.log("updated start time" + JSON.stringify(this.startTime))
+  }
+
+  updateEndTime(data: object) {
+    this.endTime = data["value"]
+    console.log("updated end time" + JSON.stringify(this.endTime))
+  }
+
+  updateStatus(data: object) {
+    this.gameActive = data["value"]
+    console.log("updated status" + JSON.stringify(this.gameActive))
+  }
+
```

Comment:
Are these functions supposed to do something? Are we putting listeners just to console.log?

Reply:
All of these functions are used to update all the game classes for each user to match. Notice how they all are setting values